community/modules/scripts/spack-install/variables.tf

```diff
@@ -58,10 +58,24 @@ variable "log_file" {
   type        = string
 }
 
+variable "data_files" {
```

Comment:
I think this would be much more clearly implemented as a `list(map(object(3 fields))` with the XOR validation of source/content. That said, this is also true of startup-script and it's a list(map) there, too.

At the very least, please cut/paste the logic for XOR. I realize that it is probably caught downstream, but I think it merits being done here.

Reply:
As mentioned in description, my preference would be to rely on underlying validation. But for the sake of moving forward I will duplicate here as I don't feel strongly.

I suspect `list(map(object(3 fields))` is typo and you meant `list(object(3 fields))`. IIUC this would require user to define one field as null in blueprint (until optionals are available in TF 1.3)

```
data_files:
- source: null
  contents: echo 'hello world'
  destination: /this/that
```

To spare user from ugliness and maintain consistency with startup script I will leave as is.